### DISCUSSION TITLE
Retrieve static files from branching endpoints

### DISCUSSION THREAD

**Discussion by capt-obvious** [fastapi/fastapi#7861] [Questions] [answered]
How can I retrieve static content with branching endpoints? I have a small project with all my templates and static files in a /dist folder. I can create a inner branch /admin/omni and correctly loaded templates, but associated static files are being pulled from /admin/dist instead of /dist. How do I prevent this behavior?

app.py
```
app = FastAPI()

app.mount('/dist', StaticFiles(directory='dist'), name='dist')
templates = Jinja2Templates(directory='dist')

@app.get("/")
async def homepage(request: Request):
return templates.TemplateResponse('homepage.html', {"request": request})

@app.get("/omni")
async def omni(request: Request):
return templates.TemplateResponse('app.html', {"request": request})

@app.get("/admin")
async def admin(request: Request):
return templates.TemplateResponse('admin.html', {"request": request})

@app.get("/admin/omni")
async def admin_omni(request: Request):
return templates.TemplateResponse('app_admin.html', {"request": request})

```

app_admin.html
```
<html>
<head>
<link rel="shortcut icon" type="image/x-icon" href="dist/favicon.68117958.ico">
<link rel="stylesheet" href="dist/app_admin.3cf53168.css">
</head>
<body>
<div id="app_admin"></div>
<script src="dist/app_admin.6744ee61.js"></script>
</body>
</html>
```

console
```
INFO:     172.22.176.1:65209 - "GET /admin/omni HTTP/1.1" 200 OK
INFO:     172.22.176.1:65209 - "GET /admin/dist/app_admin.3cf53168.css HTTP/1.1" 404 Not Found
INFO:     172.22.176.1:65208 - "GET /admin/dist/app_admin.6744ee61.js HTTP/1.1" 404 Not Found
```

**Comment by dmontagu** [comment, marked as answer]
Does changing `href="dist/...` to `href="/dist/...` and similarly for `src` in your html template work?

I think this is just related to how browsers resolve resource references, so if you need it to be relative, you might just need to structure your static files directory appropriately.

**Comment by capt-obvious** [comment]
@dmontagu that did the trick, I will have to look at my configuration for parcel to correct the build process. Thank you so much, I'm new to using python and fastapi

**Comment by dmontagu** [comment]
No problem!

Since it sounds like your question is answered, I'm going to close the issue, but please feel free to reopen it if it turns out your problem isn't quite solved, and please keep submitting questions / bug reports / etc. as you have them!

**Comment by tiangolo** [comment]
Thanks @dmontagu ! :bowing_man: :cake:
